zerver/tests/test_auth_backends.py

```diff
@@ -860,6 +860,9 @@ def social_auth_test(self, account_data_dict: Dict[str, str],
             )
             self.register_extra_endpoints(requests_mock, account_data_dict, **extra_data)
 
+            if self.AUTH_FINISH_URL == "/complete/github/" and not account_data_dict.get('avatar_url'):
+                account_data_dict['avatar_url'] = ''
```

Comment:
Can we put this github-specific code into github's `social_auth_test_finish` perhaps? Would be much cleaner than putting this kind of logic into the generic social_auth_test